Float OI (#25)
Windesheim-HBO-ICT/Leerlijn-Content-SE: content/2. Ondersteunende-informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/float/1. Float.md

---
title: 1. Float
taxonomie:
  - ib-19.2.Datatype-Primitief-Float.OI
  - ib-19.3.Datatype-Primitief-Float.OI
---

## Wat is een Float?

> Het datatype `float` is een numeriek datatype dat gebruikt wordt om decimale waarden op te slaan. Het is handig wanneer je een getal met een fractie wilt opslaan, zoals bijvoorbeeld bij meetgegevens of percentages. Een `float` in C# biedt ongeveer seven cijfers van precisie en gebruikt 32 bits voor opslag.

Author: Kan, maar dan als link naar externe bron

> Om een `float`-waarde in te voeren, gebruik je de letter `f` achter het getal, anders zal C# het getal standaard als een `double` interpreteren.

Reviewer: Dit kopje bestaat niet?

> [!TIP] Casus  

> > Stel je ontwikkelt een applicatie voor een weerstation die real-time temperatuurmetingen bijhoudt. Deze meetgegevens worden elke minuut opgevraagd en opgeslagen in `float`-variabelen om het geheugengebruik te beperken en voldoende nauwkeurigheid te behouden. Door het gebruik van het `float`-type kan de applicatie efficiënt omgaan met grote hoeveelheden meetgegevens, terwijl de nauwkeurigheid van de temperatuurmetingen behouden blijft.

Author: Tweede zin loopt beetje krom en is erg lang

Reviewer 2: Hoop dat het nu beter is

## Waarom gebruik je een `float`?

> Een `float` is een goede keuze wanneer je voldoende hebt aan een nauwkeurigheid van 7 cijfers en niet veel geheugen wilt verbruiken. In C# gebruiken floats minder geheugen dan doubles, wat nuttig kan zijn in situaties waar geheugenbeheer belangrijk is, zoals in mobiele applicaties of embedded systemen.

Author: 7 uitschrijven naar zeven. Onder de 10 altijd uitschrijven

**Voorbeeld beperking**
```C#
// Definieer pi met float en double
float piFloat = 3.1415926535f; // Opgeslagen als een float
double piDouble = 3.1415926535; // Opgeslagen als een double (nauwkeuriger)

// Toon beide waarden
Console.WriteLine("Pi als float: " + piFloat);   // Beperkte precisie
Console.WriteLine("Pi als double: " + piDouble); // Hogere precisie

// Berekening die afhankelijk is van precisie
float cirkelOppervlakteFloat = piFloat / 100f;  // Gebruik van float
double cirkelOppervlakteDouble = piDouble / 100; // Gebruik van double

// Toon oppervlaktes
Console.WriteLine("Cirkeloppervlakte met float: " + cirkelOppervlakteFloat);
Console.WriteLine("Cirkeloppervlakte met double: " + cirkelOppervlakteDouble);
```

**Resultaat**
De `float` geeft nu een minder nauwkeurig cirkel oppervlakte, doordat het eerder af moet ronden om binnen de 7 decimalen te passen.
```
Pi als float: 3.1415927
Pi als double: 3.1415926535
Cirkeloppervlakte met float: 0.03141593
Cirkeloppervlakte met double: 0.031415926535
```

---

> Volgende stap: 

Author: Verwijzing toevoegen

---
title: 1. Float
taxonomie:
  - ib-19.2.Datatype-Primitief-Float.OI
  - ib-19.3.Datatype-Primitief-Float.OI
---

## Wat is een Float?

> Het datatype `float` is een numeriek datatype dat gebruikt wordt om decimale waarden op te slaan. Het is handig wanneer je een getal met een fractie wilt opslaan, zoals bijvoorbeeld bij meetgegevens of percentages. Een `float` in C# biedt ongeveer seven cijfers van precisie en gebruikt 32 bits voor opslag [Bits in C#](Het datatype `float` is een numeriek datatype dat gebruikt wordt om decimale waarden op te slaan. Het is handig wanneer je een getal met een fractie wilt opslaan, zoals bijvoorbeeld bij meetgegevens of percentages. Een `float` in C# biedt ongeveer seven cijfers van precisie en gebruikt 32 bits voor opslag [[https://floating-point-gui.de/languages/csharp/|Bits in C#]].)

Reviewer 3: ```suggestion
Het datatype `float` is een numeriek datatype dat gebruikt wordt om decimale waarden op te slaan. Het is handig wanneer je een getal met een fractie wilt opslaan, zoals bijvoorbeeld bij meetgegevens of percentages. Een `float` in C# biedt ongeveer zeven cijfers van precisie en gebruikt 32 bits voor opslag ([[https://floating-point-gui.de/languages/csharp/|Bits in C#]]).
```

Om een `float`-waarde in te voeren, gebruik je de letter `f` achter het getal, anders zal C# het getal standaard als een -=DOUBLE=-  interpreteren.

> [!TIP] Casus  
> Stel je ontwikkelt een applicatie voor een weerstation dat elke minuut temperatuurmetingen bijhoudt en deze opslaat om het geheugen efficiënt te gebruiken en de nauwkeurigheid van de gegevens te waarborgen.

## Waarom gebruik je een Float?

> Een `float` is een goede keuze wanneer je voldoende hebt aan een nauwkeurigheid van seven cijfers en niet veel geheugen wilt verbruiken. In C# gebruiken floats minder geheugen dan doubles, wat nuttig kan zijn in situaties waar geheugenbeheer belangrijk is, zoals in mobiele applicaties of embedded systemen.

Reviewer 3: ```suggestion
Een `float` is een goede keuze wanneer je voldoende hebt aan een nauwkeurigheid van zevencijfers en niet veel geheugen wil verbruiken. In C# gebruiken floats minder geheugen dan doubles, wat nuttig kan zijn in situaties waar geheugenbeheer belangrijk is, zoals in mobiele applicaties of embedded systemen.
```

Reviewer 4: zevencijfers. lol

> [!TIP] Casus  
> Stel je ontwikkelt een mobiele app die de temperatuur en vochtigheid van verschillende ruimtes in een huis monitort. De metingen vereisen geen hoge precisie, waardoor je voor elke waarde een `float` kunt gebruiken. Dit bespaart geheugen, wat de prestaties van de app op mobiele apparaten ten goede komt, terwijl je toch voldoende nauwkeurige gegevens krijgt.

> 

Reviewer 3: ```suggestion
### Codevoorbeeld
```

```C#
float temperatuur = 23.5f;
float percentage = 78.2f;
```

## Hoe gebruik je een Float?
Je kunt een `float` gebruiken om allerlei soorten berekeningen uit te voeren, zoals optellen, aftrekken, vermenigvuldigen en delen. Houd echter rekening met mogelijke afrondingsfouten vanwege de beperkte precisie van float.

> [!TIP] Casus  
> Stel je bouwt een applicatie om de gemiddelde snelheid van een auto te berekenen. De afstand en tijd worden als `float`-waarden opgeslagen, zodat je snelheden kunt berekenen met voldoende precisie zonder veel geheugen te verbruiken.

**Mogelijke uitwerking van de casus**
```C#
float afstand = 150.5f; // afstand in kilometers 
float tijd = 2.3f; // tijd in uren float 
gemiddeldeSnelheid = afstand / tijd; 
Console.WriteLine("De gemiddelde snelheid is: " + gemiddeldeSnelheid + " km/u");
```

> ## Beperkingen van Float

Reviewer 3: ```suggestion
### Beperkingen van Float
```
En dit verplaatsen onder het hoe zit het in elkaar hoofdstuk

> De nauwkeurigheid van een `float` neemt verder af naarmate de waarde verder van nul ligt, omdat er minder ruimte overblijft voor de cijfers na de komma.

**Voorbeeld beperking**
```C#
// Definieer pi met float en double
float piFloat = 3.1415926535f; // Opgeslagen als een float
double piDouble = 3.1415926535; // Opgeslagen als een double (nauwkeuriger)

// Toon beide waarden
Console.WriteLine("Pi als float: " + piFloat);   // Beperkte precisie
Console.WriteLine("Pi als double: " + piDouble); // Hogere precisie

// Berekening die afhankelijk is van precisie
float cirkelOppervlakteFloat = piFloat / 100f;  // Gebruik van float
double cirkelOppervlakteDouble = piDouble / 100; // Gebruik van double

// Toon oppervlaktes
Console.WriteLine("Cirkeloppervlakte met float: " + cirkelOppervlakteFloat);
Console.WriteLine("Cirkeloppervlakte met double: " + cirkelOppervlakteDouble);
```

**Resultaat**
De `float` geeft nu een minder nauwkeurig cirkel oppervlakte, doordat het eerder af moet ronden om binnen de zeven decimalen te passen.
```
Pi als float: 3.1415927
Pi als double: 3.1415926535
Cirkeloppervlakte met float: 0.03141593
Cirkeloppervlakte met double: 0.031415926535
```

---

> > Volgende stap: [[content/4. Deeltaken/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/1. Float|Deeltaken]]

Reviewer 3: ```suggestion
> Volgende stap: [[content/4. Deeltaken/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/1. Float|Deeltaken float]]
```

---
title: 1. Float
taxonomie:
  - ib-19.2.Datatype-Primitief-Float.OI
  - ib-19.3.Datatype-Primitief-Float.OI
---

## Wat is een Float?
Het datatype `float` is een numeriek datatype dat gebruikt wordt om decimale waarden op te slaan. Het is handig wanneer je een getal met een fractie wilt opslaan, zoals bijvoorbeeld bij meetgegevens of percentages. Een `float` in C# biedt ongeveer zeven cijfers van precisie en gebruikt 32 bits voor opslag ([[https://floating-point-gui.de/languages/csharp/|Bits in C#]]).

Om een `float`-waarde in te voeren, gebruik je de letter `f` achter het getal, anders zal C# het getal standaard als een -=DOUBLE=-  interpreteren.

> [!TIP] Casus  
> Stel je ontwikkelt een applicatie voor een weerstation dat elke minuut temperatuurmetingen bijhoudt en deze opslaat om het geheugen efficiënt te gebruiken en de nauwkeurigheid van de gegevens te waarborgen.

> ## Hoe gebruik je een Float?

Reviewer 3: Hier mis je een kopje van "Hoe zit een Float in elkaar". Er zijn nu twee "Hoe gebruik je een Float"

Een `float` is een goede keuze wanneer je voldoende hebt aan een nauwkeurigheid van zeven cijfers en niet veel geheugen wil verbruiken. In C# gebruiken floats minder geheugen dan doubles, wat nuttig kan zijn in situaties waar geheugenbeheer belangrijk is, zoals in mobiele applicaties of embedded systemen.

> [!TIP] Casus  
> Stel je ontwikkelt een mobiele app die de temperatuur en vochtigheid van verschillende ruimtes in een huis monitort. De metingen vereisen geen hoge precisie, waardoor je voor elke waarde een `float` kunt gebruiken. Dit bespaart geheugen, wat de prestaties van de app op mobiele apparaten ten goede komt, terwijl je toch voldoende nauwkeurige gegevens krijgt.

### codevoorbeeld
```C#
float temperatuur = 23.5f;
float percentage = 78.2f;
```

### Beperkingen van Float

> float-waarden zijn minder nauwkeurig dan -=DOUBLE=--waarden, omdat ze minder bits gebruiken voor de opslag. Dit kan leiden tot afrondingsfouten bij berekeningen met meer dan zeven decimalen.

Author: Hoofdletter F
